### ISSUE TITLE
undefined is not a function

### ISSUE THREAD

**Issue by Jpunt** [visionmedia/superagent#676]
In our logging I've found an exception that I can't find the reason for. It's a classical helpful "undefined is not a function" and it seems to have something to do with Superagent:

```
Message: undefined is not a function
    Request.Request.callback in /app/node_modules/superagent/lib/node/index.js:781
    ClientRequest.[anonymous] in /app/node_modules/superagent/lib/node/index.js:736
    ClientRequest.ClientRequest.emit in events.js:107
    TLSSocket.TLSSocket.socketOnEnd in _http_client.js:300
    TLSSocket.TLSSocket.emit in events.js:129
    unknown.[anonymous] in _stream_readable.js:908
    process.process._tickCallback in node.js:355
``` 

We're making several superagent-calls to several different endpoints and I don't know at what URL or under what conditions this exception is thrown. 

Looking at the code, it seems to have something to do with a missing callback:

https://github.com/visionmedia/superagent/blob/master/lib/node/index.js#L781
```js
//...
var fn = this._callback;
//...
if (err) {
  return fn(err, res); // #781
}
//...
```

https://github.com/visionmedia/superagent/blob/master/lib/node/index.js#L736
```js
//...
req.on('error', function(err){
    // flag abortion here for out timeouts
    // because node will emit a faux-error "socket hang up"
    // when request is aborted before a connection is made
    if (self._aborted) return;
    self.callback(err); // #736
  });
//...
```

We don't have any calls without a callback. Do we have to handle exceptions like timeouts in a different/manual manner maybe?

**Comment by defunctzombie** [comment]
You shouldn't need to. Have you been able to track this down anymore?

**Comment by Jpunt** [comment]
No, I can't for the life of me figure out why this is happening. The exception is still thrown a lot and I'll try to make time to take another look at it. Any hints of what direction I should be looking?

**Comment by GDreyV** [comment]
Hi, not sure if it's the same issue but I managed to reproduce this exception. I just created request and didn't call end(). I thought that request just shouldn't be issued but it causes exception instead. The point was that I was writing the test and I didn't want to send real HTTP request. I thought not calling end() should do the trick.
So I have to call abort() and now tests hangs for a moment. It looks like request is really sent and aborting takes some time.
